c++/src/kj/async-prelude.h

```diff
@@ -27,6 +27,20 @@
 #include "exception.h"
 #include "tuple.h"
 
+// Detect whether or not we should enable kj::Promise<T> coroutine integration.
+#ifdef __has_include
+// For now, we only support the Coroutines TS.
+//
+// TODO(someday): Also support standardized C++20 Coroutines. The latest VS2019 and GCC 10 both have
+//   support, though MSVC hides it behind /std:c++latest, which brings an ICE with it.
+#if __has_include(<experimental/coroutine>) && defined(__cpp_coroutines) && __cpp_coroutines
```

Comment:
Maybe put the `__has_include` last since it presumably hits the filesystem which makes it a lot slower than the other checks.

Also, nitpick, in KJ we don't generally check `defined(X)` if we're going to then check the truthiness of `X`, since the preprocessor will default it to 0 anyawy.

Reply:
Done: https://github.com/capnproto/capnproto/pull/1104/commits/58e0eeffdfe32d033d9617cc9ea5f6cd29902431